core/src/main/scala/org/apache/spark/scheduler/ShuffleMapTask.scala

```diff
@@ -51,6 +52,7 @@ import org.apache.spark.rdd.RDD
  *                  at the same time for a barrier stage.
  */
 private[spark] class ShuffleMapTask(
+    @transient val shuffleId: Int,
```

Comment:
Do we need to add this ? We can query `DAGScheduler` given a stageId to find the shuffleId (from `ShuffleMapStage.shuffleDep,shuffleId`) - this can be done once, at `TaskSetManager` creation, and persisted as a private field, to make sure it is minimal cost.

Thoughts ?

Reply:
Make it an Option ?

Reply:
Updated to use Option.